target/linux/mvebu/image/Makefile

```diff
@@ -195,7 +196,7 @@ define Device/armada-388-clearfog-pro
   KERNEL_INSTALL := 1
   KERNEL := kernel-bin
   DEVICE_TITLE := SolidRun ClearFog Pro
-  DEVICE_PACKAGES := mkf2fs e2fsprogs swconfig kmod-fs-vfat kmod-nls-cp437 kmod-nls-iso8859-1
+  DEVICE_PACKAGES := mkf2fs e2fsprogs swconfig
```

Comment:
is ext4 support always included by default?

Reply:
Ext4 is a built-in in this target https://github.com/openwrt/openwrt/blob/adaf1cbcc8b253ea807dbe0416b4b04c33dceadf/target/linux/mvebu/config-4.9#L159.

Reply:
Have you verified this works? IIRC the clearfog u-boot loads the kernel from the vfat partition and I don't see any u-boot changes.

Also does this affect sysupgrade with keeping config? The vfat partition is used for temporary storage of the saved config, can this cause issues when upgrading from a 17.01 system?

Reply:
I don't have clearfog, so can't test it if it works there. The u-boot for clearfog has ext4 enabled and is using distro boot compatibility (set of autodetecting commands, see the bootscripts, there is no mention of fat), so should work without problems.

The sysupgrade was tested with espressobin with the same sd card image layout (without bootloader), and it went smoothly.

I'll try to ping the author of the changes which my commits touch and ask him if he can test them.

Reply:
I gave it a test, and while it boots fine with the new system,  a sysupgrade from LEDE 17.01 will fail to backup the configuration:
```
root@testname:/# sysupgrade -v /tmp/openwrt-mvebu-armada-388-clearfog-pro-sdcard.img.gz
Saving config files...
etc/config/dhcp
etc/config/dropbear
etc/config/firewall
etc/config/luci
etc/config/network
etc/config/rpcd
etc/config/system
etc/config/ubootenv
etc/config/ucitrack
etc/config/uhttpd
etc/dropbear/dropbear_rsa_host_key
etc/group
etc/hosts
etc/inittab
etc/luci-uploads/.placeholder
etc/opkg/keys/5151f69420c3f508
etc/opkg/keys/72a57f2191b211e0
etc/opkg/keys/792d9d9b39f180dc
etc/opkg/keys/9ef4694208102c43
etc/opkg/keys/b5043e70f9a75cde
etc/opkg/keys/dace9d4df16896bf
etc/opkg/keys/dd6de0d06bbd3d85
etc/passwd
etc/profile
etc/rc.local
etc/shadow
etc/shells
etc/sysctl.conf
etc/sysctl.d/local.conf
killall: watchdog: no process killed
Sending TERM to remaining processes ... uhttpd ntpd dnsmasq ubusd logd rpcd netifd odhcpd
Sending KILL to remaining processes ...
Switching to ramdisk...
Performing system upgrade...
0+155 records in
0+155 records out
mount: mounting /dev/mmcblk0p1 on /boot failed: Invalid argument
umount: can't unmount /boot: Invalid argument
Upgrade completed
Rebooting system...
(...)
root@OpenWrt:/#
```

As you can see the old system fails to mount the new /boot as it tries to mount it as vfat. This breaks keeping the configuration, which is bad IMHO.

Reply:
This didn't occur to me previously, You'r right. I'll keep the packages and modify the sysupgrade script. After the next stable is out, I'll send PR to remove fat leftovers.